app/src/main/java/de/westnordost/streetcomplete/quests/baby_changing_table/AddBabyChangingTable.java

```diff
@@ -0,0 +1,53 @@
+package de.westnordost.streetcomplete.quests.baby_changing_table;
+
+import android.os.Bundle;
+
+import javax.inject.Inject;
+
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.data.QuestImportance;
+import de.westnordost.streetcomplete.data.osm.SimpleOverpassQuestType;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment;
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;
+
+public class AddBabyChangingTable extends SimpleOverpassQuestType
+{
+	@Inject public AddBabyChangingTable(OverpassMapDataDao overpassServer)
+	{
+		super(overpassServer);
+	}
+
+	@Override
+	protected String getTagFilters()
+	{
+		return "nodes, ways with (((amenity ~ restaurant|pub|cafe|ice_cream|fuel or shop=mall or building=retail) and toilets=yes) or amenity=toilets) and !diaper";
```

Comment:
pub was suggested in the issue (I wasn't sure about that one either), and ice_cream is similar to cafe, I added it because in some cities it is indistinguishable (but yeah, they probably should be marked cafe;ice_cream).

After the push it also came to me that McDonnalds is one of the sure places one can find a changing table (at least in Poland) so I would also add fast_food to the list.

I'm not sure about limiting to named amenities is a good idea, it's a missed opportunity (that's the same issue as in opening_hours quest).